### PR TITLE
feat: Add custom_allowlist to verify_messages function

### PR DESCRIPTION
**Pull request by KTokarze**
This pull request enhances the flexibility of the `verify_messages` method in the `mfd_dmesg` module by allowing users to provide a custom allowlist of benign error messages to ignore. It also adds comprehensive unit tests to verify this new functionality and ensure correct behavior in various scenarios.

### Enhancements to allowlist functionality

* The `verify_messages` method in `base.py` now accepts a `custom_allowlist` parameter, enabling callers to specify which error messages should be ignored during verification, instead of always using the default `DMESG_WHITELIST`.
* The error filtering logic in `verify_messages` is updated to use the provided `custom_allowlist` rather than the static whitelist, making the error checking more customizable.

### Testing improvements

* Several new unit tests are added in `test_base.py` to cover cases with a custom allowlist, partial matches, and an empty allowlist, ensuring the new functionality works as intended and edge cases are handled. [[1]](diffhunk://#diff-4c585b0467f22f65904f1ac9b59361f2d80beb576cdc2fec82afbf54b43b3a60R389-R432) [[2]](diffhunk://#diff-4c585b0467f22f65904f1ac9b59361f2d80beb576cdc2fec82afbf54b43b3a60R888-R916)

### Type hinting and imports

* The type hints and imports in `base.py` are updated to support the new parameter and maintain type safety, specifically by adding `Any` to the imports.

### CODE REVIEW

File: mfd_dmesg/base.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 import datetime
 import logging
 import re
-from typing import Iterable, Optional, Union, List, Tuple, TYPE_CHECKING
+from typing import Iterable, Optional, Union, List, Tuple, TYPE_CHECKING, Any
```
[the remark this answers is not in the document]
Comment (KTokarze) [reply mfd_dmesg/base.py]: Done

File: mfd_dmesg/base.py [hunk 2]
```diff
@@ -220,9 +220,11 @@ def _check_specific_errors(self, error_msg: str) -> bool:
         else:
             return True
 
-    def verify_messages(self) -> dict:
+    def verify_messages(self, custom_allowlist: list[str | Any] = DMESG_WHITELIST) -> dict:
```
[the remark this answers is not in the document]
Comment (KTokarze) [reply mfd_dmesg/base.py]: Done

[the remark this answers is not in the document]
Comment (KTokarze) [reply mfd_dmesg/base.py]: Done

File: mfd_dmesg/base.py [hunk 3]
```diff
@@ -233,7 +235,7 @@ def verify_messages(self) -> dict:
             for error in out.splitlines():
                 is_error = True
                 if self._check_specific_errors(error):
-                    for benign_message in DMESG_WHITELIST:
+                    for benign_message in custom_allowlist:
                         if benign_message in error:
```
[the remark this answers is not in the document]
Comment (KTokarze) [reply mfd_dmesg/base.py]: Done

File: tests/unit/test_mfd_dmesg/test_base.py [hunk 4]
```diff
@@ -386,6 +386,50 @@ def test_verify_messages_no_errors(self, dmesg):
         )
         assert dmesg.verify_messages()["successful"]
 
+    def test_verify_messages_with_custom_allowlist(self, dmesg):
```
[the remark this answers is not in the document]
Comment (KTokarze) [reply tests/unit/test_mfd_dmesg/test_base.py]: Done

File: tests/unit/test_mfd_dmesg/test_base.py [hunk 5]
```diff
@@ -386,6 +386,50 @@ def test_verify_messages_no_errors(self, dmesg):
...
+        )
+        custom_allowlist = ["Couldn't get size", "MODSIGN", "unexpected notification"]
+        dmesg._connection.execute_command.return_value = ConnectionCompletedProcess(
+            return_code=0, args="command", stdout=output, stderr="stderr"
+        )
+        result = dmesg.verify_messages(custom_allowlist=custom_allowlist)
+        assert result["successful"]
+        assert result["error"] == ""
+
+    def test_verify_messages_with_custom_allowlist_partial_match(self, dmesg):
```
[the remark this answers is not in the document]
Comment (KTokarze) [reply tests/unit/test_mfd_dmesg/test_base.py]: Done

File: tests/unit/test_mfd_dmesg/test_base.py [hunk 6]
```diff
@@ -386,6 +386,50 @@ def test_verify_messages_no_errors(self, dmesg):
...
+        custom_allowlist = ["Couldn't get size"]
+        dmesg._connection.execute_command.return_value = ConnectionCompletedProcess(
+            return_code=0, args="command", stdout=output, stderr="stderr"
+        )
+        result = dmesg.verify_messages(custom_allowlist=custom_allowlist)
+        assert not result["successful"]
+        assert "MODSIGN" in result["error"]
+        assert "unexpected notification" in result["error"]
+
+    def test_verify_messages_with_empty_custom_allowlist(self, dmesg):
```
[the remark this answers is not in the document]
Comment (KTokarze) [reply tests/unit/test_mfd_dmesg/test_base.py]: Done

File: tests/unit/test_mfd_dmesg/test_base.py [hunk 7]
```diff
@@ -841,6 +885,35 @@ def test_verify_messages_no_errors(self, dmesg):
         )
         assert dmesg.verify_messages()["successful"]
 
+    def test_verify_messages_with_custom_allowlist(self, dmesg):
```
[the remark this answers is not in the document]
Comment (KTokarze) [reply tests/unit/test_mfd_dmesg/test_base.py]: Done

File: tests/unit/test_mfd_dmesg/test_base.py [hunk 8]
```diff
@@ -841,6 +885,35 @@ def test_verify_messages_no_errors(self, dmesg):
...
+        )
+        custom_allowlist = ["Couldn't get size", "MODSIGN"]
+        dmesg._connection.execute_command.return_value = ConnectionCompletedProcess(
+            return_code=0, args="command", stdout=output, stderr="stderr"
+        )
+        result = dmesg.verify_messages(custom_allowlist=custom_allowlist)
+        assert result["successful"]
+        assert result["error"] == ""
+
+    def test_verify_messages_with_custom_allowlist_partial_match(self, dmesg):
```
[the remark this answers is not in the document]
Comment (KTokarze) [reply tests/unit/test_mfd_dmesg/test_base.py]: Done

File: mfd_dmesg/base.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (KTokarze) [reply mfd_dmesg/base.py]: Done

**Review by adrianlasota — CHANGES_REQUESTED** [review]
I don't see reason, why we cannot extent constants list

File: mfd_dmesg/base.py [hunk 9]
```diff
@@ -220,11 +220,15 @@ def _check_specific_errors(self, error_msg: str) -> bool:
         else:
             return True
 
-    def verify_messages(self) -> dict:
+    def verify_messages(self, custom_allowlist: Optional[Iterable[str]] = None) -> dict:
```
Comment (adrianlasota) [inline mfd_dmesg/base.py]: use new typehints
Iterable[str] | None

**Review by adrianlasota — CHANGES_REQUESTED** [review]

File: mfd_dmesg/base.py (hunk 9 above)
Comment (mchromin) [inline mfd_dmesg/base.py]:
```suggestion
    def verify_messages(self, custom_allowlist: Iterable[str] | None = None) -> dict:
```

**Review by mchromin — CHANGES_REQUESTED** [review]

File: mfd_dmesg/base.py (hunk 9 above)
Comment (KTokarze) [reply mfd_dmesg/base.py]: Done

**Review by mchromin — APPROVED** [review]
